File: Tutorial5-1Dagger-Android/src/main/java/com/example/tutorial5_1dagger_android/di/MainActivitySubComponent.java
package com.example.tutorial5_1dagger_android.di;

import com.example.tutorial5_1dagger_android.MainActivity;

import dagger.Subcomponent;
import dagger.android.AndroidInjector;

/**
 * SubComponents can be removed in dagger-android with @ContributesAndroidInjector
 * if SubComponent and it's Builder has NO methods
 */
@Subcomponent
public interface MainActivitySubComponent extends AndroidInjector<MainActivity> {

    @Subcomponent.Builder
    abstract class Builder extends AndroidInjector.Builder<MainActivity> {

    }
}

File: Tutorial5-4Dagger-Android-MultipleActivitiesAndFragments/src/main/java/com/smarttoolfactory/tutorial5_4dagger_android_multipleactivitiesandfragments/main/MainActivitySubComponent.kt
package com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.main

import com.example.tutorial5_3dagger_android_multipleactivitiesandfragments.di.scope.ActivityScope
import com.smarttoolfactory.tutorial5_4dagger_android_multipleactivitiesandfragments.di.FragmentContributorModule
import dagger.Subcomponent
import dagger.android.AndroidInjector


@ActivityScope
@Subcomponent(modules = [MainActivityModule::class, FragmentContributorModule::class])
interface MainActivitySubComponent : AndroidInjector<MainActivity> {

    @Subcomponent.Builder
    abstract class Builder : AndroidInjector.Builder<MainActivity>()

}